src/python/pants/backend/codegen/tasks/simple_codegen_task.py

```diff
@@ -310,7 +310,14 @@ def name(cls):
       return 'global'
 
     def execute_codegen(self, targets):
-      self._task.execute_codegen(targets)
+      try:
+        self._task.execute_codegen([target])
+      except ZeroDivisionError as ex:
```

Comment:
I don't think we need to catch ZeroDivisionError explicitly.  You can remove this except case.